File: 0118-pascals-triangle/0118-pascals-triangle.cpp
class Solution {
public:
    vector<int> generaterow(int numrows){

        vector<int> ans;

        ans.push_back(1);

        int temp = 1;

        for(int col = 1 ; col < numrows ; col++){

            temp = temp * (numrows - col);
            temp /= col;
            ans.push_back(temp);

        }

        return ans;
    }
    vector<vector<int>> generate(int numrows){

        vector<vector<int>> v;

        for(int i = 1 ; i <= numrows ; i++){
 
            v.push_back(generaterow(i));

        }

        return v ;
    }
};

File: 0118-pascals-triangle/0118-pascals-triangle.java
class Solution {
    public List<Integer> gen(int row){
        int temp = 1;
        List<Integer> v = new ArrayList<>();
        v.add(1);

        for(int col = 1 ; col < row ; col++){
            temp = temp * (row - col);
            temp /= col;
            v.add(temp);
        }
        return v;
    }
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> ans = new ArrayList();
        for(int i = 1 ; i <= numRows ; i++){
            ans.add(gen(i));
        }
        return ans;
    }
}